src/main/java/com/google/collegeplanner/data/Course.java

```diff
@@ -0,0 +1,220 @@
+   */
+  private String courseId;
+  /*
+   * name represents the full name of the course.
+   * Example, Academic Writing
+   */
+  private String name;
+  /*
+   * semester represents the 4 digit year followed by
+   * the two digit month when the course starts. Example, 202008
+   */
+  private String semester;
+  /*
+   * credits represents the number of credits the course is worth.
+   */
+  private int credits;
+  /*
+   * departmentId represents the 4 character ID of the
+   * department that the class falls into. Example, ENGL
+   */
+  private String departmentId;
+  /*
+   * description represents the description of
+   * the class found on the course catalog.
+   */
+  private String description;
+  /*
+   * gradingMethod represents the list of possible grading
+   * methods students can opt for. Example, regular, pass-fail.
+   */
+  private String[] gradingMethod;
+  /*
+   * corequisites represents the corequisite classes for this
+   * course. Example, BIO106 is the lab for BIO101 and
+   * must be taken together
+   */
+  private String corequisites;
+  /*
+   * prerequisites represents the prerequisite classes for
+   * this course. Example, MATH101, MATH201, and
+   * either MATH105 or MATH115.
+   */
+  private String prerequisites;
+  /*
+   * restrictions represents the restrictions placed on the
+   * registration of this course. Example, This
+   * class is only avaiable to incoming Freshman.
+   */
+  private String restrictions;
+  /*
+   * additionalInfo represents the additional information
+   * for this course found on the course catalog.
+   */
+  private String additionalInfo;
+  /*
+   *creditGrantedFor represents the classes this course
+   * grants credit for. Example, THET285 grants credit for COMM107
+   */
+  private String creditGrantedFor;
+
+  public Course(String courseId, String name, String semester, int credits, String departmentId,
+      String description, String[] gradingMethod, String corequisites, String prerequisites,
+      String restrictions, String additionalInfo, String creditGrantedFor) {
+    this.courseId = courseId;
+    this.name = name;
+    this.semester = semester;
+    this.credits = credits;
+    this.departmentId = departmentId;
+    this.description = description;
+    this.gradingMethod = gradingMethod;
+    this.corequisites = corequisites;
+    this.prerequisites = prerequisites;
+    this.restrictions = restrictions;
+    this.additionalInfo = additionalInfo;
+    this.creditGrantedFor = creditGrantedFor;
+  }
+
+  public Course(JSONObject json) {
+    this((String) json.get("course_id"), (String) json.get("name"), (String) json.get("semester"),
+        (int) json.get("credits"), (String) json.get("dept_id"), (String) json.get("description"),
+        null, (String) json.get("coreqs"), (String) json.get("prereqs"),
+        (String) json.get("restrictions"), (String) json.get("additional_info"),
+        (String) json.get("credit_granted_for"));
```

Comment:
All the requirements / restrictions are actually nested inside of an array called `relationships`: https://beta.umd.io/#operation/getCourses. So that needs to be manually unpacked like you did with `gradingMethod`.